attestation.go

```diff
@@ -20,11 +20,11 @@ type Attestation struct {
 	Signature   Signature  `serix:"2,mapKey=signature"`
 }
 
-func NewAttestation(api API, block *ProtocolBlock) *Attestation {
+func NewAttestation(api API, block *Block) *Attestation {
 	return &Attestation{
 		API:         api,
-		BlockHeader: block.BlockHeader,
-		BlockHash:   lo.PanicOnErr(block.Block.Hash()),
+		BlockHeader: block.Header,
+		BlockHash:   lo.PanicOnErr(block.Body.Hash()),
```

Comment:
Rename to BodyHash

Reply:
Implemented all your suggestions, thanks!